.github/get_latest_index.py

```diff
@@ -0,0 +1,95 @@
+import os
+import re
+import requests
+import sys
+import uuid
+from google.cloud import bigquery
+
+class IDCIndexManager:
+    def __init__(self, project_id):
+        print("Initializing IDCIndexManager...")
+        self.project_id = project_id
+        self.client = bigquery.Client(project=project_id)
+
+    def get_latest_idc_release_version(self, view_id):
+        print("Getting latest IDC release version...")
+        view = self.client.get_table(view_id)
+        latest_idc_release_version=int(re.search(r"idc_v(\d+)", view.view_query).group(1))
+        return latest_idc_release_version
```

Comment:
After browsing https://console.cloud.google.com/marketplace/product/bigquery-public-data/nci-idc-data?project=just-experience-186619

I was able to browser through the table `bigquery-public-data.idc_current.dicom_all_view`:

![image](https://github.com/ImagingDataCommons/idc-index/assets/219043/3d59dbfb-b630-4a1b-8a87-976228cf0906)

That said, I couldn't see the verion information we are exacting there. Do you know where are these organized ?

Reply:
JC, it is a long story. `idc_current` BQ dataset contains views defined by queries that point to the latest version of the IDC release, which in turn are stored in the datasets named as `bigquery-public-data.idc_v<release_number>`, so currently the latest one is `idc_v17`. There is a known bug with the underlying mechanism from Google that we use that does not expose the actual query behind the view. In the past, you could find it in the Details section of the view. Unfortunately, we do not have any estimate from Google on when this would be fixed. Here's the tracker, but unfortunately it is not public - we will try to make it public: https://issuetracker.google.com/issues/324112186.

Fortunately, this is not a stopper, since we can list all of the `idc_v*` datasets to get the latest version. See relevant discussion in https://github.com/ImagingDataCommons/idc-index/issues/11#issuecomment-1946308527 - @vkt1414 is going to work on this after we are done with the current time-sensitive activity that is a priority (should be done in 1-2 weeks).

cc: @bcli4d for awareness

Reply:
Thanks for the explanation, that makes sense, and we can indeed see the versioned table listed:

![image](https://github.com/ImagingDataCommons/idc-index/assets/219043/20e4f2d9-f33b-47a8-96f5-980c9ab8c7e9)

Reply:
@jcfr I actually found a better way to get the version info. Sorry I'm behind but I'm planning to work on this gha soon. 

![image](https://github.com/ImagingDataCommons/idc-index/assets/115020590/39d0a150-dab0-4c74-a419-b765a32eb4b4)

Reply:
@vkt1414 yes, this is indeed a more elegant and I agree a better approach!